src/algebra/ring/basic.lean

```diff
@@ -101,6 +116,18 @@ class semiring (α : Type u) extends add_comm_monoid α, monoid_with_zero α, di
 section semiring
 variables [semiring α]
 
+@[priority 100] -- see Note [lower instance priority]
+instance semiring.to_non_unital_non_assoc_semiring : non_unital_non_assoc_semiring α :=
+{ ..‹semiring α› }
+
+@[priority 100] -- see Note [lower instance priority]
+instance semiring.non_unital_semiring : non_unital_semiring α :=
+{ ..‹semiring α› }
+
+@[priority 100] -- see Note [lower instance priority]
+instance semiring.non_assoc_semiring : non_assoc_semiring α :=
+{ ..‹semiring α› }
```

Comment:
I wonder; does lowering these to `99` solve the elaborator issues you point out below?

Reply:
I tried setting the priority to `99` on all three instances here but the elaborator issues remained.